src/checkout-buttons/strategies/googlepay/googlepay-checkoutcom-button-strategy.spec.ts

```diff
@@ -0,0 +1,216 @@
+    let formPoster: FormPoster;
+    let checkoutButtonOptions: CheckoutButtonInitializeOptions;
+    let paymentMethod: PaymentMethod;
+    let paymentProcessor: GooglePayPaymentProcessor;
+    let checkoutActionCreator: CheckoutActionCreator;
+    let requestSender: RequestSender;
+    let store: CheckoutStore;
+    let strategy: GooglePayButtonStrategy;
+    let walletButton: HTMLAnchorElement;
+
+    beforeEach(() => {
+        paymentMethod = getPaymentMethod();
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        requestSender = createRequestSender();
+
+        checkoutActionCreator = checkoutActionCreator = new CheckoutActionCreator(
+            new CheckoutRequestSender(requestSender),
+            new ConfigActionCreator(new ConfigRequestSender(requestSender))
+        );
+
+        paymentProcessor = createGooglePayPaymentProcessor(
+            store,
+            new GooglePayCheckoutcomInitializer()
+        );
+
+        formPoster = createFormPoster();
+
+        strategy = new GooglePayButtonStrategy(
+            store,
+            formPoster,
+            checkoutActionCreator,
+            paymentProcessor
+        );
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(paymentProcessor, 'initialize')
+            .mockReturnValue(Promise.resolve());
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        jest.spyOn(formPoster, 'postForm')
+            .mockReturnValue(Promise.resolve());
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        jest.spyOn(paymentProcessor, 'deinitialize');
+
+        container.appendChild(walletButton);
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    it('creates an instance of GooglePayButtonStrategy', () => {
+        expect(strategy).toBeInstanceOf(GooglePayButtonStrategy);
+    });
+
+    describe('#initialize()', () => {
+
+        describe('Payment method exist', () => {
+
+            it('Creates the button', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('Validates if strategy is been initialized', async () => {
```

Comment:
Just reword the test to make sense
```suggestion
            it('Validates if strategy has been initialized', async () => {
```

Reply:
fixed